import1/dbus_test.go

```diff
@@ -0,0 +1,184 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package import1
+
+import (
+	"io"
+	"net/http"
+	"os"
+	"testing"
+)
+
+const (
+	importPrefix = "importd-test-"
+)
+
+const (
+	rawURL = "https://dl.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.raw.xz"
+	tarURL = "http://old-releases.ubuntu.com/releases/ubuntu-core/releases/14.10/release/ubuntu-core-14.10-core-amd64.tar.gz"
+)
+
+func TestImportTar(t *testing.T) {
+	prepareImage(t, "image.tar.xz", tarURL)
+
+	conn, err := New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f, err := os.Open("tmp/image.tar.xz")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.ImportTar(f, importPrefix+"ImportTar", true, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestImportRaw(t *testing.T) {
+	prepareImage(t, "image.raw.xz", rawURL)
+
+	conn, err := New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f, err := os.Open("tmp/image.raw.xz")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.ImportRaw(f, importPrefix+"ImportRaw", true, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestExportTar(t *testing.T) {
+	prepareImage(t, "image.tar.xz", tarURL)
+
+	conn, err := New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f, err := os.Create("tmp/image-export.tar.xz")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.ExportTar(importPrefix+"ImportTar", f, "xz")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestExportRaw(t *testing.T) {
+	prepareImage(t, "image.raw.xz", tarURL)
+
+	conn, err := New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f, err := os.Create("tmp/image-export.raw.xz")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.ExportRaw(importPrefix+"ImportRaw", f, "xz")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestPullTar(t *testing.T) {
+	conn, err := New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.PullTar(tarURL, importPrefix+"PullTar", "no", true)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+func TestPullRaw(t *testing.T) {
+	conn, err := New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.PullRaw(rawURL, importPrefix+"PullRaw", "no", true)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestListAndCancelTransfers(t *testing.T) {
+	conn, err := New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	transfers, err := conn.ListTransfers()
+	if err != nil {
+		t.Error(err)
+	}
+	if len(transfers) < 1 {
+		t.Error("transfers length is not correct")
+	}
+
+	for _, v := range transfers {
+		err = conn.CancelTransfer(v.Id)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+func prepareImage(t *testing.T, name, url string) {
+	err := os.MkdirAll("tmp", os.ModeDir|0755)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	filepath := "tmp/" + name
+
+	_, err = os.Stat(filepath)
+	if err == nil {
+		return
+	}
+	if !os.IsNotExist(err) {
+		t.Fatal(err)
+	}
+
+	res, err := http.Get(url)
```

Comment:
Do we need a full-image and a remote download here? Can we maybe just have a small `.raw.xz` and `.tar.gz` under `/fixtures/`?

Reply:
Well, I used to think systemd will check validity of the image, but I found I was wrong. I will add fixtures instead.